src/main/java/org/sagebionetworks/bridge/adherence/WeeklyAdherenceReportWorkerProcessor.java

```diff
@@ -0,0 +1,162 @@
+import org.sagebionetworks.bridge.rest.model.App;
+import org.sagebionetworks.bridge.rest.model.Study;
+import org.sagebionetworks.bridge.rest.model.StudyPhase;
+import org.sagebionetworks.bridge.rest.model.WeeklyAdherenceReport;
+import org.sagebionetworks.bridge.sqs.PollSqsWorkerBadRequestException;
+import org.sagebionetworks.bridge.worker.ThrowingConsumer;
+import org.sagebionetworks.bridge.workerPlatform.bridge.BridgeHelper;
+import org.sagebionetworks.bridge.workerPlatform.bridge.PagedResourceIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+@Component("WeeklyAdherenceReportWorker")
+public class WeeklyAdherenceReportWorkerProcessor implements ThrowingConsumer<JsonNode> {
+    private static final Logger LOG = LoggerFactory.getLogger(WeeklyAdherenceReportWorkerProcessor.class);
+
+    static final Set<StudyPhase> ACTIVE_PHASES = ImmutableSet.of(DESIGN, RECRUITMENT, IN_FLIGHT);
+    static final int PAGE_SIZE = 100;
+    static final long THREAD_SLEEP_INTERVAL = 200L;
+    
+    private BridgeHelper bridgeHelper;
+    
+    private ClientManager clientManager;
+
+    @Autowired
+    public final void setBridgeHelper(BridgeHelper bridgeHelper) {
+        this.bridgeHelper = bridgeHelper;
+    }
+    
+    @Autowired
+    public final void setClientManager(ClientManager clientManager) {
+        this.clientManager = clientManager;
+    }
+    
+    @Override
+    public void accept(JsonNode node) throws Exception {
+        WeeklyAdherenceReportRequest request;
+        try {
+            request = DefaultObjectMapper.INSTANCE.treeToValue(node, WeeklyAdherenceReportRequest.class);
+        } catch (IOException e) {
+            throw new PollSqsWorkerBadRequestException("Error parsing request: " + e.getMessage(), e);
+        }
+        Stopwatch requestStopwatch = Stopwatch.createStarted();
+        try {
+            if (request.getAppId() != null || request.getStudyId() != null) {
+                LOG.info("Limiting weekly adherence report caching to appId=" + request.getAppId() + ", studyId="
+                        + request.getStudyId());
+            }
+            process(request);
+        } finally {
+            LOG.info("Weekly adherence report caching took " + requestStopwatch.elapsed(TimeUnit.SECONDS) + " seconds");
+        }
+    }
+    
+    private void process(WeeklyAdherenceReportRequest request) throws Exception {
+        ForWorkersApi workersApi = clientManager.getClient(ForWorkersApi.class);
+        
+        List<App> appList = null;
+        if (request.getAppId() != null) {
+            appList = ImmutableList.of( bridgeHelper.getApp(request.getAppId()) );
```

Comment:
We should add a small unit test to verify that we filter appIDs if an appID is specified.

Reply:
Added verification that we don't call for an app when not specified.